library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -569,7 +616,9 @@ protected void replay(long windowId)
               if (inputStream != null) {
                 closeFile(inputStream);
               }
-              processedFiles.add(recoveryEntry.file);
+              synchronized (processedFiles.getClass()) {
```

Comment:
I tried a similar one using Collections.synchronizedSet(new HashSet<String>()). And others have similar problems as described below:

When I ran the tests I see several errors like:
Class cannot be created (missing no-arg constructor): java.util.Collections SynchronizedSet
When the code is trying to do a Kryo copy:
LineByLineFileInputOperator initialState = new Kryo().copy(oper);

Searched online and the reason is the Collections.synchronizedSet() doesn't have a no-arg constructor. Since I cannot modify this class, I have to specifically lock on each access to this processedFiles and do something like:
synchronized (processedFiles.getClass())